src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/util/ByteBufUtil.kt

```diff
@@ -20,8 +20,191 @@ package com.exactpro.th2.conn.dirty.tcp.core.util
 
 import io.netty.buffer.ByteBuf
 import io.netty.buffer.Unpooled
+import io.netty.buffer.search.AbstractSearchProcessorFactory.newKmpSearchProcessorFactory
+import java.nio.charset.Charset
+import kotlin.text.Charsets.UTF_8
 
 fun ByteBuf.asExpandable(): ByteBuf = when (maxCapacity()) {
     Int.MAX_VALUE -> this
     else -> Unpooled.wrappedBuffer(this, Unpooled.buffer())
-}
\ No newline at end of file
+}
+
+fun ByteBuf.requireReadable(fromIndex: Int, toIndex: Int) {
+    require(fromIndex < toIndex) {
+        "fromIndex must be less than toIndex: $fromIndex..$toIndex"
+    }
+
+    require(fromIndex >= readerIndex() && toIndex <= writerIndex()) {
+        "Range is outside of readable bytes: $fromIndex..$toIndex"
+    }
+}
+
+fun ByteBuf.requireReadable(index: Int) {
+    require(index in readerIndex() until writerIndex()) {
```

Comment:
Maybe replace to a condition with great and less operations instead of the IntRange object allocation

Reply:
There will be no allocation in runtime. The code above will look like this:
```java
private static final void requireReadable(ByteBuf $this$requireReadable, int index) {
   boolean var5;
   label17: {
      int var10000 = $this$requireReadable.readerIndex();
      int var10002 = $this$requireReadable.writerIndex();
      if (var10000 <= index) {
         if (var10002 > index) {
            var5 = true;
            break label17;
         }
      }

      var5 = false;
   }

   boolean var2 = var5;
   if (!var2) {
      int var3 = false;
      String var4 = "Index is outside of readable bytes: " + index;
      throw (Throwable)(new IllegalArgumentException(var4.toString()));
   }
}
```